lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -738,13 +737,8 @@ def add_decision_manager_fields(xml, options)
       def add_payment_solution(xml, payment_method)
         return unless network_tokenization?(payment_method)
 
-        case payment_method.source
-        when :network_token
-          payment_solution = NT_PAYMENT_SOLUTION[payment_method.brand]
-          xml.tag! 'paymentSolution', payment_solution if payment_solution
-        when :apple_pay, :google_pay
-          xml.tag! 'paymentSolution', @@wallet_payment_solution[payment_method.source]
-        end
+        payment_solution = payment_method.try(:network_token?) ? NT_PAYMENT_SOLUTION[payment_method.brand] : @@wallet_payment_solution[payment_method.source]
```

Comment:
Minor detail. I think we could skip the `.try` here, since the early return guarantees that it is a `NetworkTokenizationCreditCard`. Same thing in the `add_payment_network_token` method.

Reply:
NetworkTokenizationCreditCard can be Network Token, ApplePay or GooglePay and `payment_method.try(:network_token?) ` ensures that it is a Network Token.

Reply:
Every `NetworkTokenizationCreditCard` should have the method `#network_token?` available. I think we could call `payment_method.network_token?` directly instead of using `try`.

Reply:
@Buitragox you're right, I've update it